### PR TITLE
implement `mmap` with basic flags for aarch64

### PR DESCRIPTION
**Pull request by ken4647**
This PR implements syscalls  as follows:
1. `sys_mmap`: support flags `MAP_PRIVATE`, `MAP_SHARED`, `MAP_ANONYMOUS`, `MAP_FILE`, `MAP_FIXED`.
2. `sys_munmap`
3. `sys_mprotect`
4. `sys_msync`: only support `MS_SYNC` temporarily.
5. `sys_mremap`
6. `sys_pread64`: the accessibility of file is temporarily unchecked.
7. `sys_pwrite64`: the accessibility of file is temporarily unchecked.

At the same time, it also implements the ability to swap memory. To enable memory swapping, you need to create a file named "swap.raw" in the root directory. When free memory is lower than 4MB, `PageFault` will swap memory instead of alloc memory to prevent the panic . **This logic is very simple, and it will possibily be smarter in the furture** .  

`sys_mmap` only support `aarch64`, so it will be unserviceable temporarily for `x86` or `riscv` until next PR. Besides, `HugePage` is also unsupported for now.

### CODE REVIEW

**Review by coolyjg — COMMENTED** [review]
Some bugs should be solved.

File: api/ruxos_posix_api/src/imp/mmap.rs [hunk 1]
```diff
@@ -6,91 +6,914 @@
...
+    }
+}
+
+struct TrapHandlerImpl;
+
+#[crate_interface::impl_interface]
+impl ruxhal::trap::TrapHandler for TrapHandlerImpl {
+    fn handle_page_fault(vaddr: usize, cause: i64) -> i64 {
+        let vma_map = VMA_MAP.lock();
+        if pte_map_check(VirtAddr::from(vaddr)).is_ok() {
```
[the remark this answers is not in the document]
Comment (coolyjg) [reply api/ruxos_posix_api/src/imp/mmap.rs:225]: For multi-thread

**Comment by AuYang261** [comment]
It seems that the exsiting funciton [`sys_pread`](https://github.com/syswonder/ruxos/blob/dev/api/ruxos_posix_api/src/imp/fs.rs#L595rl) is the same as the newly added function `sys_pread64`. Are they neccessary to be merged into one?

**Comment by ken4647** [comment]
> It seems that the exsiting funciton [`sys_pread`](https://github.com/syswonder/ruxos/blob/dev/api/ruxos_posix_api/src/imp/fs.rs#L595rl) is the same as the newly added function `sys_pread64`. Are they neccessary to be merged into one?

ok，I think it is okay to merged into one. You can raise your PR with sys_pread64 and sys_pwrite64 with nothing changed, then I will adapt for yours to make this PR fine.

**Comment by ken4647** [comment]
> It seems that the exsiting funciton [`sys_pread`](https://github.com/syswonder/ruxos/blob/dev/api/ruxos_posix_api/src/imp/fs.rs#L595rl) is the same as the newly added function `sys_pread64`. Are they neccessary to be merged into one?

`sys_pread64` and `sys_pread` are not the same, but it is my prime target to use the api to read in different offset. `sys_pwrite` is unimplemented, so i didn't notice `sys_pread` then? I will merge them into one, and unify the sys_pread as sys_pread.(As i actually find the implement of pread64 is the pread.)

**Comment by coolyjg** [comment]
`sys_pread` is not a standard syscall name, and should be removed (or later). Keep `sys_pread64` only.

**Pull request closed without merging**